pennylane/ops/qubit/state_preparation.py

```diff
@@ -172,23 +183,25 @@ class StatePrep(StatePrepBase):
     ndim_params = (1,)
     """int: Number of dimensions per trainable parameter of the operator."""
 
-    def __init__(self, state: TensorLike, wires: WiresLike, id: Optional[str] = None):
-        super().__init__(state, wires=wires, id=id)
-        state = self.parameters[0]
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        state: TensorLike,
+        wires: WiresLike,
+        pad_with=None,
+        normalize=False,
```

Comment:
```suggestion
        pad_with: Optional[complex] = None,
        normalize: Optional[bool] = False,
```

Reply:
Please do not upgrade the UI in this PR, we can have a follow-up PR for all these upgrades, that should be approved by Product.